packages/grid/x-grid/src/tests/rows.XGrid.test.tsx

```diff
@@ -342,5 +342,40 @@ describe('<XGrid /> - Rows', () => {
       const isVirtualized = apiRef!.current!.getState().containerSizes!.isVirtualized;
       expect(isVirtualized).to.equal(false);
     });
+
+    describe('Pagination', () => {
+      it('should render only the pageSize ', () => {
+        render(<TestCaseVirtualization pagination pageSize={32} />);
+        const gridWindow = document.querySelector('.MuiDataGrid-window')!;
+        gridWindow.scrollTop = 10e6; // scroll to the bottom
+        gridWindow.dispatchEvent(new Event('scroll'));
+
+        const lastCell = document.querySelector(
+          '[role="row"]:last-child [role="cell"]:first-child',
+        )!;
+        expect(lastCell.textContent).to.equal('31');
+        const totalHeight = apiRef!.current!.getState().containerSizes?.totalSizes.height!;
```

Comment:
This seems to be the only reason for testing `XGrid` and not `DataGrid`. I wonder if for the tests, we shouldn't expose the apiRef so we can have as much tests as possible on `DataGrid`.

Reply:
Testing DataGrid and XGrid is the same as they both use GridComponent under the hood

Reply:
As far as we know, the plan is to have them as two different components, where XGrid uses extra "plugins". So we should optimize now for this known future. 

In such configuration, I wonder if it's better to tests the version of the component with the fewer plugins, DataGrid, or with the most XGrid. I guess it depend.

- what's most likely to happen: a missing plugin to implement the behavior or an extra plugin breaking the behavior? No idea.
- what's most used? DataGrid
- what's more mission critical? XGrid

Reply:
Let's keep that in mind.
- We could run each test on each implementation, and we should expect some test to fail. But that would be quite big to setup. I don't think we are there yet, more important things to focus on
- We could stand on the more feature we have the more difficult it is to integrate so we test on XGrid and if something should be turned off for dataGrid, we check that it works as expected. 
- We check everything on DataGrid and if we need apiRef or testing an xgrid feature we switch. Kind of bad DX IMO

Anyway I think those are very core feature. So it would be difficult for the grid to work without atm. 
IMHO, I think we should take a pragmatic approach and merge as this PR fixes a few regressions.

Reply:
I was leaving a general note, agree that it's outside of the scope of the changes here.